src/serlo-editor/plugins/multimedia/editor.tsx

```diff
@@ -29,6 +31,20 @@ export function MultimediaEditor(props: MultimediaProps) {
     selectHasFocusedDescendant(state, multimedia.id)
   )
 
+  // inside of box plugin don't allow video and geogebra as multimedia children
+  const typesOfAncestors = useAppSelector((state) =>
+    selectAncestorPluginTypes(state, props.id)
+  )
+  const hasBoxAnchestor = typesOfAncestors?.includes(EditorPluginType.Box)
+  const filteredPlugins = hasBoxAnchestor
+    ? config.allowedPlugins.filter(
+        (plugin) =>
+          ![EditorPluginType.Video, EditorPluginType.Geogebra].includes(
+            plugin as EditorPluginType
+          )
+      )
+    : config.allowedPlugins
+
```

Comment:
@hejtful what do you think, how terrible is this for performance? 😸 
I would memo it but I'm not sure how to recalculate it reliably if someone moves the plugin to a new parent.

Reply:
I guess it will impact performance, but not sure how much without profiling 😄 

We could reduce the impact by not using `useAppSelector`, and instead checking for `filteredPlugins` value when the modal opens. This way, the selector won't impact re-renders, and I think it's a safe trigger for re-evaluation of this value, because it can't change while the modal is open. I hope I managed to describe what I mean in an understandable way 😄 If not, I can try to rephrase.

Reply:
Hm yes, I think I understood. 
The setting will likely move into to toolbar at some point so I guess than it would not work any more. 
🤔

Reply:
Thanks for the help!